week1/hhplus-tdd-java/src/main/java/io/hhplus/tdd/point/PointService.java

```diff
@@ -0,0 +1,59 @@
+package io.hhplus.tdd.point;
+
+import io.hhplus.tdd.database.PointHistoryTable;
+import io.hhplus.tdd.database.UserPointTable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class PointService {
+
+    private static final Logger log = LoggerFactory.getLogger(PointService.class);
+
+    private final UserPointTable userPointTable;
+    private final PointHistoryTable pointHistoryTable;
+
+    @Autowired
+    PointService(UserPointTable userPointTable, PointHistoryTable pointHistoryTable) {
+        this.userPointTable = userPointTable;
+        this.pointHistoryTable = pointHistoryTable;
+    }
+
+    public UserPoint point(long id) {
+        UserPoint userPoint = userPointTable.selectById(id);
+        log.debug("userPoint: {}", userPoint);
+        return userPoint;
+    }
+
+    public List<PointHistory> history(long id) {
+        List<PointHistory> historyList = pointHistoryTable.selectAllByUserId(id);
+        log.info("history의 개수: {}", historyList.size());
+        return historyList;
+    }
+
+    public UserPoint charge(long id, long amount) throws Exception {
+        UserPoint userPoint = userPointTable.selectById(id);
+        if (userPoint.point() + amount >= Integer.MAX_VALUE) {
+            throw new Exception();
```

Comment:
```suggestion
            throw new PointOverflowException();
```

예외에 대해서 더 구체적인 예외로 분리하면 좋을것 같습니다
Custom Exception 을 정의하는게 가장 좋을것 같고 그게 아니라면 적당한 표준 예외를 넣어주시는것도 좋을것 같습니다!

```java
public class InsufficientBalanceException extends RuntimeException {}
public class PointOverflowException extends RuntimeException {}
```

Reply:
확실히 예외를 명확하게 정의하는게 좋을 것 같습니다!